src/services/user-group/services/course-copy-service.js

```diff
@@ -109,6 +112,10 @@ class CourseShareService {
 		this.app = app;
 	}
 
+	setup(app) {
+		this.app = app;
```

Comment:
I think you should only set the app once, either in the constructor or during setup